lib/src/Uncertainty/Distribution/FisherSnedecorFactory.cxx

```diff
@@ -45,6 +45,20 @@ FisherSnedecorFactory * FisherSnedecorFactory::clone() const
 
 /* Here is the interface that all derived class must implement */
 
+/* Algorithm associated with the method of moments */
+FisherSnedecor FisherSnedecorFactory::buildMethodOfMoments(const Sample & sample) const
+{
+  const Scalar mu = sample.computeMean()[0];
+  if (mu <= 1.0) throw InvalidArgumentException(HERE) << "Error: cannot estimate a FisherSnedecor distribution based on a sample with sample mean less than 1 using the method of moments.";
+  const Scalar sigma2 = sample.computeCovariance()(0,0);
+  if (sigma2 == 0.0) throw InvalidArgumentException(HERE) << "Error: cannot estimate a FisherSnedecor distribution based on a constant sample using the method of moments.";
+  const Scalar d2 = 2*mu/(mu-1);
+  const Scalar d1 = 2*std::pow(d2,2)*(d2-2)/(std::pow(d2-2,2)*(d2-4)*sigma2-2*std::pow(d2,2));
```

Comment:
Please ad some spaces around operators in your formula. Some compilers will complain if you take an integral power of a foating point value, it is more robust to change constants from their integral writing into their floating point writing: `std::pow(d2-2,2)` becomes `std::pow(d2 - 2.0, 2.0)`

Reply:
Ok, it makes sense, given that this amounts to x^y=exp(y*log(x)) when x is a floating point. I fixed this.